### ISSUE TITLE
brew install ant fails because of missing dependency to bcel

### ISSUE THREAD

**Issue by britter** [Homebrew/homebrew-core#44608]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [x] are reporting a bug others will be able to reproduce and not asking a question or requesting software. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh. To get software added or changed in Homebrew please file a [Pull Request](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)
- [x] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, Homebrew/homebrew-core formula (not cask) on macOS? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a Linux problem please file this issue at https://github.com/Homebrew/linuxbrew-core/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [x] ran `brew update` and can still reproduce the problem?
- [x] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [x] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?

```
~ > brew gist-logs ant
https://gist.github.com/df11f8a58bfd13eb8c4ffc50ae74a766
```

- [x] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:

- What you were trying to do (and why)

Tried to install Apache Ant by running `brew install ant`

- What happened (include command output)

Installation failed with the following output:

```
~ > brew install ant
==> Downloading https://www.apache.org/dyn/closer.cgi?path=ant/binaries/apache-ant-1.10.7-bin.tar.xz
Already downloaded: /Users/bene/Library/Caches/Homebrew/downloads/eff485c9fc5bf11232b26b3ae7f6c062a4edd8cd0e3d8a4e5a7f549521050ecb--apache-ant-1.10.7-bin.tar.xz
==> Downloading https://www.apache.org/dyn/closer.cgi?path=ant/ivy/2.4.0/apache-ivy-2.4.0-bin.tar.gz
Already downloaded: /Users/bene/Library/Caches/Homebrew/downloads/d7c4a713694cabcfdaa178db886b9f04febb4655388accfcdacfb4dc509c4205--apache-ivy-2.4.0-bin.tar.gz
==> Downloading https://www.apache.org/dyn/closer.cgi?path=commons/bcel/binaries/bcel-6.3.1-bin.tar.gz
==> Downloading from http://mirror.synyx.de/apache/commons/bcel/binaries/bcel-6.3.1-bin.tar.gz

curl: (22) The requested URL returned error: 404 Not Found
Trying a mirror...
==> Downloading https://www-eu.apache.org/dist/commons/bcel/binaries/bcel-6.3.1-bin.tar.gz

curl: (22) The requested URL returned error: 404 Not Found
Trying a mirror...
==> Downloading https://www-us.apache.org/dist/commons/bcel/binaries/bcel-6.3.1-bin.tar.gz

curl: (22) The requested URL returned error: 404 Not Found
Error: An exception occurred within a child process:
  DownloadError: Failed to download resource "ant--bcel"
Download failed: https://www-us.apache.org/dist/commons/bcel/binaries/bcel-6.3.1-bin.tar.gz
```

- What you expected to happen

Apache ant should be installed on my system without error.

- Step-by-step reproduction instructions (by running `brew install` commands)

1. run `brew install ant`

**Comment by britter** [comment]
I think the ant formula needs to updated to the latest Commons BCEL release, which is 6.4.0 or the formular needs to use the [archive mirrors](https://archive.apache.org/dist/commons/bcel/binaries/) instead of the dist mirror because the archives will keep old versions but the dist mirror will only have the latest version.

**Comment by marsavela** [comment]
Nobody has dared to solve this yet?

**Comment by tedmetal** [comment]
Seems link for bcel 6.3.1 is no longer valid. 
Pull Request Added to update bcel to 6.4.0
https://github.com/Homebrew/homebrew-core/pull/44623

**Comment by codeskyblue** [comment]
For quickly fix, the best way is modify ant Formula

```
brew edit ant
```

Change the formula according to https://github.com/Homebrew/homebrew-core/pull/44623/files

**Comment by SMillerDev** [comment]
It's already fixed in #44581
